pkg/manifests/manifests.go

```diff
@@ -1353,9 +1357,137 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
 		}
 	}
 
+	if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs != nil {
+		p.Spec.AdditionalAlertManagerConfigs = &v1.SecretKeySelector{
+			Key: AdditionalAlertmanagerConfigSecretKey,
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: AdditionalAlertmanagerConfigSecretName,
+			},
+		}
+		p.Spec.Secrets = append(p.Spec.Secrets, getAlertmanagerTLSSecretsName(f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs)...)
+	}
+
 	return p, nil
 }
 
+func (f *Factory) AdditionalAlertManagerConfigs() (*v1.Secret, error) {
+	alertmanagerConfigs := f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs
+	if len(alertmanagerConfigs) == 0 {
+		return nil, nil
+	}
+
+	cfgs := []yaml2.MapSlice{}
+	for _, alertmanagerConfig := range alertmanagerConfigs {
+		cfg := yaml2.MapSlice{}
+		cfg = append(cfg, yaml2.MapSlice{
```

Comment:
you probably want to skip injecting the fields into the YAML if the values are empty